storm-core/src/jvm/org/apache/storm/trident/TridentTopology.java

```diff
@@ -277,6 +277,35 @@ public Stream join(List<Stream> streams, List<Fields> joinFields, Fields outFiel
               outFields);
     }
 
+    public Stream joinSQL(Stream s1, Fields joinFields1, Stream s2, Fields joinFields2, Fields outFields) {
```

Comment:
Why are we exposing new apis in TridentTopology and what extra functionality does joinSQL provide ? It may confuse users if they see both Join and JoinSQL. IMO, we should have only `join` in the trident apis. The SQL specific parts should be private and may be defined outside of TridentTopology if its specific to SQL implementation.